### ISSUE TITLE
[Feature request] Firewall Enhancement

### ISSUE THREAD

**Issue by Ilithy** [Gedsh/InviZible#60]
Hi, I use your great application for Tor and I2P access (and DNSCrypt with Adguard's DNS), but it prevents me from using my usual firewall, which is very powerful, would it be possible to modify the firewall implemented in InviZible Pro, to add some useful features/UI?

1: The firewall is clean, but no explanation is given on how to use the interface; it might be interesting to add tooltips appearing when you hold your finger on a button, then a tooltip is shown explaining me the functionality of the button.

2: Would it be possible to add a tab for authorized applications, and one for blocked applications?

3: Would it be great to have a filtering log displayed, i.e., see all Internet access requests (and target) for processes ?
Something like this:
![Screenshot_20201215-123454](https://user-images.githubusercontent.com/36798218/102210091-09004900-3ed2-11eb-9f53-90b0d318abb2.jpg)

![Screenshot_20201215-123510](https://user-images.githubusercontent.com/36798218/102210132-14537480-3ed2-11eb-97ae-b3ad555dcf45.jpg)

_(The screenshots shows the Adguard firewall for Android)_

4: is it possible to add an option to automatically allow/block access of newly installed applications to WiFi and cellular networks?

5: add a sort option: sorted by last (Internet) access.

These are only suggestions for improvement, it would make the firewall easier to use, and more functional; but I understand that it is probably impossible to develop everything, thank you for your consideration on these questions.

**Comment by Gedsh** [comment]
Hi. Thank you for the suggestions!

> 1: The firewall is clean, but no explanation is given on how to use the interface; it might be interesting to add tooltips appearing when you hold your finger on a button, then a tooltip is shown explaining me the functionality of the button.

I will add.

> 2: Would it be possible to add a tab for authorized applications, and one for blocked applications?

Please show me an example. What is it for?

> 3: Would it be great to have a filtering log displayed, i.e., see all Internet access requests (and target) for processes ?

Currently, you can see such a log in the DNS tab. But not so pretty. I plan to change its interface over time.

> 4: is it possible to add an option to automatically allow/block access of newly installed applications to WiFi and cellular networks?

I have planned this feature for the next beta.

> 5: add a sort option: sorted by last (Internet) access.

It seems useless to me.

**Comment by Ilithy** [comment]
> Hi. Thank you for the suggestions!
> 
> > 1: The firewall is clean, but no explanation is given on how to use the interface; it might be interesting to add tooltips appearing when you hold your finger on a button, then a tooltip is shown explaining me the functionality of the button.
> 
> I will add.
> 
> > 2: Would it be possible to add a tab for authorized applications, and one for blocked applications?
> 
> Please show me an example. What is it for?
> 
> > 3: Would it be great to have a filtering log displayed, i.e., see all Internet access requests (and target) for processes ?
> 
> Currently, you can see such a log in the DNS tab. But not so pretty. I plan to change its interface over time.
> 
> > 4: is it possible to add an option to automatically allow/block access of newly installed applications to WiFi and cellular networks?
> 
> I have planned this feature for the next beta.


> It seems useless to me.

**Comment by Ilithy** [comment]
Hi;
My apologies, the closing was accidental.

>I will add.

Thanks you for this

>Please show me an example. What is it for?

Of course, here is an example of how this is displayed (for my example, it's still Adguard's firewall):

_Request display selection screen_
![Screenshot_20201216-080336](https://user-images.githubusercontent.com/36798218/102320060-170aa400-3f7c-11eb-846c-492333bcc326.jpg)

_Selective display only of blocked queries (here without blacklist selection)_
![Screenshot_20201216-080525](https://user-images.githubusercontent.com/36798218/102320546-bb8ce600-3f7c-11eb-94d7-777d591a638d.jpg)

This allows you to select the type of request to display (excluding some), it's quite handy when I want to check the targets of requests from my Android applications on my devices (or attempts, in the case of blocked or blacklisted requests), I don't need to search, I only display what I'm looking for (blacklisted, whitelisted, blocked etc...)

>Currently, you can see such a log in the DNS tab. But not so pretty. I plan to change its interface over time.

I understand that this must take a lot of time, thank you for paying attention.

>I have planned this feature for the next beta.

Huge! Thanks a lot

>It seems useless to me.

it's true, it's not very useful, I like to have a lot of sorting methods :)

**Comment by Gedsh** [comment]
> Of course, here is an example of how this is displayed (for my example, it's still Adguard's firewall)

Got it. You talked about the requests log.

Thank you for the suggestions! A fresh look is very helpful.

**Comment by Ilithy** [comment]
>Got it. You talked about the requests log.

 Yes, indeed I wasn't explicit. I got tangled up talking to you about "tabs", sorry.

>Thank you for the suggestions! A fresh look is very helpful.

Thanks for thinking about it.

**Comment by Yquas** [comment]
I would like number 5 too, in some way. Not the last internet request, that's not that importnant. But I'd like to be able to sort which apps have some access and which have none. For example for when the phone has 100+ system apps and you allowed some connections to like five apps, you'd want to find them easily. But right now one needs to look carefully through the entire list. And maybe it'd be possible to sort by the amount of allowed connections? Like that on the top you see apps with all four green, then three, two, one, zero. What do you say, @Gedsh?

**Comment by Gedsh** [comment]
>  I'd like to be able to sort which apps have some access and which have none.

I can move the applications that do not have Internet access to the end of the list.

> And maybe it'd be possible to sort by the amount of allowed connections?

I don't think this is a good idea. It will be unclear how to find the application if there are too many factors affecting the sort order.

**Comment by Yquas** [comment]
@Gedsh I agree that it would be a bit weird than. But moving the apps without ny permission down is a good idea!

**Comment by tommykarate** [comment]
These are good, I wish to know if I can block a whole address so no app can access it like facebook.com or google.com?

**Comment by Gedsh** [comment]
@tommykarate You can use the DNSCrypt blacklist for this purpose. MENU -> DNSCrypt Settings -> Blacklist

**Comment by tommykarate** [comment]
Sorry, I'm a noob. Never mind. I found the blacklist. Awesome app!

**Comment by tommykarate** [comment]
Oops didn't see your reply! Thank you! This is a great app!
